### PR TITLE
[Index, IndeBundle] allow condition renderers to be registered dynamically

### PR DESCRIPTION
**Pull request by dpfaffenbauer**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | yes
| Fixed tickets | #711 

This allows to register new `ConditionRenderer` and therefore a more dynamic approach to add new Conditions.

This deprecates the `MysqlRenderer` which was actually supposed to be used internally only anyway.

### CODE REVIEW

File: src/CoreShop/Bundle/IndexBundle/Condition/MysqlRenderer.php [hunk 1]
```diff
@@ -12,237 +12,42 @@
...
-use CoreShop\Component\Index\Condition\NotLikeCondition;
-use CoreShop\Component\Index\Condition\NotMatchCondition;
-use CoreShop\Component\Index\Condition\RangeCondition;
+use CoreShop\Component\Index\Condition\ConditionRendererInterface;
 use CoreShop\Component\Index\Condition\RendererInterface;
-use Pimcore\Db;
+use CoreShop\Component\Index\Worker\WorkerInterface;
 
+/**
+ * @deprecated MysqlRenderer is deprecated since 2.0.0, please use the CoreShop\Component\Index\Condition˜ConditionRendererInterface instead
```
Comment (solverat) [inline src/CoreShop/Bundle/IndexBundle/Condition/MysqlRenderer.php]: typo

File: src/CoreShop/Bundle/IndexBundle/Condition/MysqlRenderer.php [hunk 2]
```diff
@@ -12,237 +12,42 @@
...
-     * @return string
+     * Condition constructor.
      */
-    protected function renderPrefix($prefix = null)
+    public function __construct()
     {
-        if (null === $prefix) {
-            return '';
-        }
+        $this->mysqlWorker = \Pimcore::getContainer()->get('coreshop.index.worker.mysql');;
```
Comment (solverat) [inline src/CoreShop/Bundle/IndexBundle/Condition/MysqlRenderer.php]: typo

File: src/CoreShop/Bundle/IndexBundle/Condition/MysqlRenderer.php (hunk 1 above)
Comment (dpfaffenbauer) [reply src/CoreShop/Bundle/IndexBundle/Condition/MysqlRenderer.php]: fixed

File: src/CoreShop/Bundle/IndexBundle/Condition/MysqlRenderer.php (hunk 2 above)
Comment (dpfaffenbauer) [reply src/CoreShop/Bundle/IndexBundle/Condition/MysqlRenderer.php]: fixed

**Pull request merged**